frontend/src/App.tsx

```diff
@@ -215,15 +229,20 @@ function appStateReducer(state: CoveyAppState, update: CoveyAppUpdate): CoveyApp
       nextState.gameStarted = true;
       break;
     case 'disconnect':
+      state.socket?.emit('finishGame', state.myPlayerID, -1, true);
       state.socket?.disconnect();
+      closedInstructions = false;
       return defaultAppState();
     case 'toggleQuit':
       nextState.toggleQuit = !state.toggleQuit;
       break;
     case 'exitMaze':
       nextState.toggleQuit = false;
-      // TODO: have this call Player.giveUp() in the backend
-      // state.socket?.emit('giveUp');
+      nextState.gameInfo.gameStatus = 'noGame';
+      closedInstructions = false;
+      break;
+    case 'toggleGameStarted':
```

Comment:
also forgive me for bringing this up late but is there a reason we chose to have both a `gameStatus` and `gameStarted`?

Reply:
gameStarted means the instructions have been closed and the timer has started, while the 'inGame' in game status just means you're inside of the maze.  Also, when we go to 'noGame', we don't know what the previous state was, so if we go from 'invitePending' to 'noGame', teleport would be called (if we get rid of gameStarted).

Reply:
Could we not have just added an additional gameStatus?

Reply:
We probably could?  I can go in and try to refactor it

Reply:
we probably have to revisit anywhere where we expected inGame and expect inGame or gameStarted